java-analyzer-bundle.core/src/main/java/io/konveyor/tackle/core/internal/SampleDelegateCommandHandler.java

```diff
@@ -50,6 +50,43 @@ private void waitForJavaSourceDownloads() {
     // to filter.
     private static SearchPattern mapLocationToSearchPatternLocation(int location, String query) throws Exception {
         //TODO: #21 Normalize queries and/or verify for each location.
+
+        if (query.contains("(") || query.contains(")")) {
+            // We know that this is a list of things to loook for, broken by a | command. We should get this intra string and create an OR search pattern for each one. 
+            // ex ((FileWriter|FileReader|PrintStream|File|PrintWriter|RandomAccessFile))
+            var startListIndex = query.indexOf("(");
+            var endListIndex = query.indexOf(")");
+            logInfo("Start index: " + startListIndex + " end index: " + endListIndex);
+            var startQuery = query.substring(0, startListIndex);
+            var endQuery = query.substring(endListIndex+1, query.length());
+
+            // This sshould strip the ( ) chars
+            var listString = query.substring(startListIndex+1, endListIndex);
+            logInfo("start query: " + startQuery + " end query: " + endQuery + " list String: " + listString);
+            var list = listString.split("\\|");
```

Comment:
I think this whole block above can be substituted with this:
```java
Matcher m = Pattern.compile("([\\w]+)").matcher(query);
List<String> listString = m.results().map(MatchResult::group).collect(Collectors.toList());
```

Reply:
@pranavgaikwad @fabianvf, which do we prefer? I don't care but I want to make sure that everyone is onboard with using more of the lambda stuff.

Reply:
sticking with substring for now.